source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -248,24 +233,17 @@ SplitRequestPtr MGETRequest::create(Router& router, Common::Redis::RespValuePtr&
   std::vector<Common::Redis::RespValue> responses(request_ptr->num_pending_responses_);
   request_ptr->pending_response_->asArray().swap(responses);
 
-  std::vector<Common::Redis::RespValue> values(2);
-  values[0].type(Common::Redis::RespType::BulkString);
-  values[0].asString() = "get";
-  values[1].type(Common::Redis::RespType::BulkString);
-  Common::Redis::RespValue single_mget;
-  single_mget.type(Common::Redis::RespType::Array);
-  single_mget.asArray().swap(values);
-
-  for (uint64_t i = 1; i < incoming_request->asArray().size(); i++) {
+  Common::Redis::RespValueSharedPtr base_request = std::move(incoming_request);
+  for (uint64_t i = 1; i < base_request->asArray().size(); i++) {
     request_ptr->pending_requests_.emplace_back(*request_ptr, i - 1);
     PendingRequest& pending_request = request_ptr->pending_requests_.back();
 
-    single_mget.asArray()[1].asString() = incoming_request->asArray()[i].asString();
-    ENVOY_LOG(debug, "redis: parallel get: '{}'", single_mget.toString());
-    const auto route = router.upstreamPool(incoming_request->asArray()[i].asString());
+    const auto route = router.upstreamPool(base_request->asArray()[i].asString());
     if (route) {
+      Common::Redis::RespValue single_mget(base_request,
```

Comment:
This is a composite array right? Can it be fully specified for clarity like you do below?

Reply:
Creating the composite array this way would allocate the composite array once and keep it const.  I'll add a comment that this is composite array